ci/scripts/run_ci_local.sh

```diff
@@ -0,0 +1,128 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+case "$1" in
+    "" )
+        STAGES=("bash")
+        ;;
+    "all" )
+        STAGES=("checks" "build-clang" "build-gcc" "test-clang" "test-gcc" "codecov" "docs" "benchmark" "conda")
+        ;;
+    "build" )
+        STAGES=("build-clang" "build-gcc")
+        ;;
+    "test" )
+        STAGES=("test-clang" "test-gcc")
+        ;;
+    "checks" | "build-clang" | "build-gcc" | "test" | "test-clang" | "test-gcc" | "codecov" | "docs" | "benchmark" | \
+    "conda" | "bash" )
+        STAGES=("$1")
+        ;;
+    * )
+        echo "Error: Invalid argument \"$1\" provided. Expected values: \"all\", \"checks\", \"build\", " \
+             "\"build-clang\", \"build-gcc\", \"test\", \"test-clang\", \"test-gcc\", \"codecov\"," \
+             "\"docs\", \"benchmark\", \"conda\" or \"bash\""
+        exit 1
+        ;;
+esac
+
+# CI image doesn't contain ssh, need to use https
+function git_ssh_to_https()
+{
+    local url=$1
+    echo $url | sed -e 's|^git@github\.com:|https://github.com/|'
+}
+
+MRC_ROOT=${MRC_ROOT:-$(git rev-parse --show-toplevel)}
+
+GIT_URL=$(git remote get-url origin)
+GIT_URL=$(git_ssh_to_https ${GIT_URL})
+
+GIT_UPSTREAM_URL=$(git remote get-url upstream)
+GIT_UPSTREAM_URL=$(git_ssh_to_https ${GIT_UPSTREAM_URL})
+
+GIT_BRANCH=$(git branch --show-current)
+GIT_COMMIT=$(git log -n 1 --pretty=format:%H)
+
+BASE_LOCAL_CI_TMP=${BASE_LOCAL_CI_TMP:-${MRC_ROOT}/.tmp/local_ci_tmp}
+CONTAINER_VER=${CONTAINER_VER:-230920}
+CUDA_VER=${CUDA_VER:-11.8}
+DOCKER_EXTRA_ARGS=${DOCKER_EXTRA_ARGS:-""}
+
+BUILD_CONTAINER="nvcr.io/ea-nvidia-morpheus/morpheus:mrc-ci-build-${CONTAINER_VER}"
+TEST_CONTAINER="nvcr.io/ea-nvidia-morpheus/morpheus:mrc-ci-test-${CONTAINER_VER}"
+
+# These variables are common to all stages
+BASE_ENV_LIST="--env LOCAL_CI_TMP=/ci_tmp"
+BASE_ENV_LIST="${BASE_ENV_LIST} --env GIT_URL=${GIT_URL}"
+BASE_ENV_LIST="${BASE_ENV_LIST} --env GIT_UPSTREAM_URL=${GIT_UPSTREAM_URL}"
+BASE_ENV_LIST="${BASE_ENV_LIST} --env GIT_BRANCH=${GIT_BRANCH}"
+BASE_ENV_LIST="${BASE_ENV_LIST} --env GIT_COMMIT=${GIT_COMMIT}"
+BASE_ENV_LIST="${BASE_ENV_LIST} --env PARALLEL_LEVEL=$(nproc)"
+BASE_ENV_LIST="${BASE_ENV_LIST} --env CUDA_VER=${CUDA_VER}"
+BASE_ENV_LIST="${BASE_ENV_LIST} --env SKIP_CONDA_ENV_UPDATE=${SKIP_CONDA_ENV_UPDATE}"
+
+for STAGE in "${STAGES[@]}"; do
+    # Take a copy of the base env list, then make stage specific changes
+    ENV_LIST="${BASE_ENV_LIST}"
+
+    if [[ "${STAGE}" =~ benchmark|clang|codecov|gcc ]]; then
+        if [[ "${STAGE}" =~ "clang" ]]; then
+            BUILD_CC="clang"
+        elif [[ "${STAGE}" == "codecov" ]]; then
```

Comment:
Question: one of these uses regex, the other is an exact string match. Is there significance to that?

Reply:
Yes, the first is checking for either the "build-clang" or "test-clang" stage, however there is only a single "codecov" stage.